pkg/scalers/azure/azure_aad_workload_identity.go

```diff
@@ -117,41 +127,48 @@ func getScopedResource(resource string) string {
 }
 
 type ADWorkloadIdentityConfig struct {
-	ctx        context.Context
-	IdentityID string
-	Resource   string
+	ctx                   context.Context
+	IdentityID            string
+	IdentityTenantID      string
+	IdentityAuthorityHost string
+	Resource              string
 }
 
-func NewAzureADWorkloadIdentityConfig(ctx context.Context, identityID, resource string) auth.AuthorizerConfig {
-	return ADWorkloadIdentityConfig{ctx: ctx, IdentityID: identityID, Resource: resource}
+func NewAzureADWorkloadIdentityConfig(ctx context.Context, identityID, identityTenantID, identityAuthorityHost, resource string) auth.AuthorizerConfig {
+	return ADWorkloadIdentityConfig{ctx: ctx, IdentityID: identityID, IdentityTenantID: identityTenantID, IdentityAuthorityHost: identityAuthorityHost, Resource: resource}
 }
 
 // Authorizer implements the auth.AuthorizerConfig interface
 func (aadWiConfig ADWorkloadIdentityConfig) Authorizer() (autorest.Authorizer, error) {
 	return autorest.NewBearerAuthorizer(NewAzureADWorkloadIdentityTokenProvider(
-		aadWiConfig.ctx, aadWiConfig.IdentityID, aadWiConfig.Resource)), nil
+		aadWiConfig.ctx, aadWiConfig.IdentityID, aadWiConfig.IdentityTenantID, aadWiConfig.IdentityAuthorityHost, aadWiConfig.Resource)), nil
 }
 
-func NewADWorkloadIdentityCredential(identityID string) (*azidentity.WorkloadIdentityCredential, error) {
+func NewADWorkloadIdentityCredential(identityID, identityTenantID string) (*azidentity.WorkloadIdentityCredential, error) {
 	options := &azidentity.WorkloadIdentityCredentialOptions{}
 	if identityID != "" {
 		options.ClientID = identityID
 	}
+	if identityTenantID != "" {
+		options.TenantID = identityTenantID
+	}
```

Comment:
I think that we have to propagate the AuthorityHost until here and set `options.Cloud.ActiveDirectoryAuthorityHost` as the value is recovered from the env var inside the SDK. Honestly, I'm not 100% sure but I guess that if we set it for using rest SDK, we should use it also here.

Reply:
@JorTurFer I think I follow what you are suggesting... The `AZURE_AUTHORITY_HOST` environment variable is set by the [azure-workload-identity admission mutating webhook](https://github.com/Azure/azure-workload-identity/blob/91933470e433358e1e37d84f4cc1dce4b7813ad7/pkg/webhook/webhook.go#L51). The init() function in azure_aad_workload_identity.go will load the default authority host from that environment variable and override it if a user sets an `identityAuthorityHost` value in the `podIdentity` object. 

Instead of setting the tenant and authority host overrides separately, I think I should move the `identityAuthorityHost` check to only happen if `identityTenantId` is also set. Like this:

```suggestion
	if identityTenantID != "" {
		options.TenantID = identityTenantID
		
		// override the authority host only if provided and tenant id is provided
		if identityAuthorityHost != "" {
			authorityHost = identityAuthorityHost
		}
	}
```

That way we'll only be overriding the authority host if the tenant id is also set and reduce the potential for a mismatch between tenant and the cloud it is homed in.

Reply:
> The init() function in azure_aad_workload_identity.go will load the default authority host from that environment variable and override it if a user sets an identityAuthorityHost value in the podIdentity object.

Yes, but this field is only used for the rest client xD

> Instead of setting the tenant and authority host overrides separately, I think I should move the identityAuthorityHost check to only happen if identityTenantId is also set. Like this:

I think that it's a good approach, but in the other hand, we won't notify the error to the user. I mean, if the user sets the AAD endpoint and not the tenantID, it's an error that we should show to the user as. Furthermore, I can be missing something because I have no experience in other clouds than public, but I guess that KEDA won't work either so maybe raising an error could be the best option (and also adding it as validation during admission). 
WDYT?